test/cpp/api/misc.cpp

```diff
@@ -92,6 +132,7 @@ TEST(DeterministicTest, CanSetDeterministic) {
 }
```

Comment:
I didn't realize this when I reviewed this previously, but the fact that these tests are toggling global state (deterministic in the test case) means they are not safe for parallel execution; e.g., if someone decided they wanted to parallelize the gtest suite, these wouldn't work.

Reply:
Do you think we should just remove these tests? I think that would be acceptable since this functionality is being exercised indirectly in the python tests.

Reply:
Yeah, I would just get rid of the C++ side tests